src/psyclone/dynamo0p3.py

```diff
@@ -6285,133 +7348,84 @@ def __init__(self, kern, parent):
-                                 entity_decls=["ncell_2d"]))
 
     def field_vector(self, argvect):
-        '''add the field vector associated with the argument 'argvect' to the
-        argument list '''
-        undf_name = get_fs_undf_name(argvect.function_space)
-        from psyclone.f2pygen import DeclGen
+        '''Add the field vector associated with the argument 'argvect' to the
+        argument list.
+
+        :param argvect: the corresponding kernel argument.
+        :type argvect:  :py:class:`psyclone.dynamo0p3.DynKernelArgument`
+        '''
         # the range function below returns values from
         # 1 to the vector size which is what we
         # require in our Fortran code
         for idx in range(1, argvect.vector_size+1):
             text = (argvect.name + "_" +
                     argvect.function_space.mangled_name +
                     "_v" + str(idx))
-            intent = argvect.intent
-            decl = DeclGen(self._parent, datatype="real",
-                           kind="r_def", dimension=undf_name,
-                           intent=intent, entity_decls=[text])
-            self._parent.add(decl)
             if self._first_arg:
                 self._first_arg = False
-                self._first_arg_decl = decl
             self._arglist.append(text)
 
     def field(self, arg):
-        '''add the field associated with the argument 'arg' to the argument
-        list'''
-        from psyclone.f2pygen import DeclGen
-        undf_name = get_fs_undf_name(arg.function_space)
+        '''
+        Add the field associated with the argument 'arg' to the argument list.
+
+        :param arg: the kernel argument (field).
+        :type arg: :py:class:`psyclone.dynamo0p3.DynKernelArgument`
+        '''
         text = arg.name + "_" + arg.function_space.mangled_name
-        intent = arg.intent
-        decl = DeclGen(self._parent, datatype="real",
-                       kind="r_def", dimension=undf_name,
-                       intent=intent, entity_decls=[text])
-        self._parent.add(decl)
-        if self._first_arg:
-            self._first_arg = False
-            self._first_arg_decl = decl
         self._arglist.append(text)
 
     def stencil_unknown_extent(self, arg):
         '''add stencil information associated with the argument 'arg' if the
         extent is unknown'''
-        from psyclone.f2pygen import DeclGen
-        name = arg.name + "_stencil_size"
-        self._parent.add(DeclGen(self._parent, datatype="integer", intent="in",
-                                 entity_decls=[name]))
+        name = DynStencils.dofmap_size_name(arg)
         self._arglist.append(name)
 
     def stencil_unknown_direction(self, arg):
         '''add stencil information associated with the argument 'arg' if the
         direction is unknown'''
-        from psyclone.f2pygen import DeclGen
         name = arg.name+"_direction"
```

Comment:
Why is this name explicitly created here but others use the name space manager via DynStencils?

Reply:
Oh yes, I don't know! I've added a new method to DynStencils and updated this method to use that.